build.csx

```diff
@@ -68,12 +74,62 @@ targets.Add("default", DependsOn("gitlink", "unit", "integ", "spec", "approve",
 
 targets.Add("outputDirectory", () => Directory.CreateDirectory(outputDirectory));
 
+targets.Add("coverityDirectory", () => Directory.CreateDirectory(coverityDirectory));
+
 targets.Add("logsDirectory", () => Directory.CreateDirectory(logsDirectory));
 
 targets.Add("testsDirectory", () => Directory.CreateDirectory(testsDirectory));
 
 targets.Add("build", DependsOn("clean", "outputDirectory", "restore"), () => RunMsBuild("Build"));
 
+targets.Add(
+    "coverity",
+    DependsOn("clean", "coverityDirectory", "restore"),
+    () =>
+    {
+        var packagesDirectoryOption = $"/p:NuGetPackagesDirectory={packagesDirectory}";
+        Cmd(
+            "cov-build",
+            $@"--dir {coverityResultsDirectory} ""{msBuild}"" {solution} /target:Build /p:configuration=Release /nr:false /verbosity:minimal /nologo /fl /flp:LogFile=artifacts/logs/Coverity-Build.log;Verbosity=Detailed;PerformanceSummary {packagesDirectoryOption}");
+
+        var version = GetVersion();
+        var coverityToken = Environment.GetEnvironmentVariable("COVERITY_TOKEN");
+        var coverityEmail = Environment.GetEnvironmentVariable("COVERITY_EMAIL");
+        var repoCommitId = Environment.GetEnvironmentVariable("APPVEYOR_REPO_COMMIT");
+
+        var coverityZipFile = coverityDirectory + "/coverity.zip";
+        Cmd("7z", $"a -r {coverityZipFile} {coverityResultsDirectory}");
+
+        using (var client = new HttpClient())
+        {
+            client.Timeout = TimeSpan.FromMinutes(20);
+
+            var form = new MultipartFormDataContent();
+            form.Add(new StringContent(coverityToken), @"""token""");
+            form.Add(new StringContent(coverityEmail), @"""email""");
+            form.Add(new StringContent(version), @"""version""");
+            form.Add(new StringContent($"Build {version} ({repoCommitId})"), @"""description""");
+
+            using (var fileStream = new FileStream(coverityZipFile, FileMode.Open, FileAccess.Read))
+            {
+                StreamContent formFileField = new StreamContent(fileStream);
+
+                form.Add(formFileField, @"""file""", "coverity.zip");
+
+                Console.WriteLine("Uploading coverity scan...");
+                var postTask = client.PostAsync(coverityProjectUrl, form);
+                try
+                {
+                    postTask.Wait();

```

Comment:
Sorry about this. I've been meaning to look into async tasks for simple-targets-csx: https://github.com/adamralph/simple-targets-csx/issues/87